src/main/java/core/basesyntax/dao/impl/StorageDaoImpl.java

```
@@ -0,0 +1,28 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.db.Storage;
+import java.util.Map;
+
+public class StorageDaoImpl implements StorageDao {
+
+    @Override
+    public void addFruitQuantity(String fruit, int quantity) {
+        Storage.putFruitQuantity(fruit, getFruitQuantity(fruit) + quantity);
```

addFruitQuantity does not validate inputs. It should check that `fruit` is not null/empty and that `quantity` is non-negative, and throw an informative RuntimeException if the arguments are invalid. The task description explicitly requires handling null parameters and invalid input data (for example: “Null parameters”) so DAO-layer methods must validate their parameters as well: Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance. For example: ... 3. Null parameters . Also follow the guidance to rethrow informative RuntimeException instead of ignoring exceptions .